gamemode/modules/rafts/sh_raft_class.lua

```diff
@@ -83,6 +80,8 @@ function raftMeta:GetSpawnPosition( ply )
     local _, highestPiece = next( self.pieces )
 
     for _, raftPiece in pairs( self.pieces ) do
+        if not IsValid( raftPiece ) then continue end
```

Comment:
perhaps you could put alot of this in a local function then the loop would become something like  
```lua
for _, raftPiece in pairs( self.pieces ) do
    if IsValid( raftPiece ) then
        local pos = getValidSpawnPos( ply, raftPiece )
        if pos then return pos end

        if raftPiece:GetPos().z > highestPiece:GetPos().z then
            highestPiece = raftPiece
        end
    end
end
```
(im not saying I require this, if you like the continue thats fine)